_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,244 @@
+
+toc::[]
+
+== OpenID Connect
+https://openid.net/developers/how-connect-works/[OpenID Connect] is an identity layer on top of the OAuth 2.0 protocol that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID provider. A future release of WildFly will allow the user to configure additional scope values to request specific claim values.
+
+== Prerequisites
+We can deploy this application using Red Hat OpenShift https://developers.redhat.com/developer-sandbox[Developer Sandbox]. There are other https://www.redhat.com/en/technologies/cloud-computing/openshift/try-it[options] available for running OpenShift. Please refer to this https://docs.wildfly.org/29/Getting_Started_on_OpenShift.html[getting started guide] to learn more about WildFly on OpenShift. We will be making use of the following tools for this walkthrough: 
+
+* https://docs.docker.com/[Docker] (you can also use https://docs.podman.io/en/latest/[Podman])  
+* https://developers.redhat.com/developer-sandbox[Developer Sandbox]
+* https://docs.openshift.com/container-platform/4.11/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* https://quay.io/repository/[Quay]
+* https://helm.sh/[Helm]
+
+== Example Application
+For this walkthrough, we will be making use of the https://github.com/PrarthonaPaul/elytron-examples/tree/elytron-oidc-client-scope/elytron-oidc-client-scope[elytron-oidc-client-scope] project which makes use of the ability to configure additional scope values for OpenID Connect. First, clone the `elytron-example` repo locally and navigate to the `elytron-oidc-client-scope` project:
+```
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+cd elytron-examples/elytron-oidc-client-scope
+```
+
+== Log Into OpenShift Cluster
+
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the OpenShift CLI:
+```
+oc login -u myUserName
+```
+
+Alternatively, to log in using the API token, navigate to Developer Sandbox, click on your username at the top right corner, copy your login command and paste it on your terminal. 
+The command will be of this format: 
+```
+oc login --token=myToken --server=myServerUrl
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift. 
+
+== Start Red Hat Single Sign On 
+Developer Sandbox gives us access to the enterprise version of Red Hat's OpenID provider, Red Hat Single Sign On (RH-SSO). On the topology view of OpenShift, click on the `Add to Project` button at the top left and add the version of RH-SSO that is compatible with the version of OpehShift you are running and wait for it to finish deploying. Please refer to the https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.2/html/red_hat_single_sign-on_for_openshift/introduction[RH-SSO for OpenShift Documentation] to learn more about it.
```

Comment:
Please note that RH-SSO will fail to complete running. 
Please follow this to resolve this issue: https://access.redhat.com/solutions/4874991
[This](https://access.redhat.com/solutions/5318441) post mentions that the issue has been reported.